src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.Accessor.cs

```diff
@@ -106,25 +167,22 @@ private bool ChecksumMatches(TKey key, Checksum checksum, CancellationToken canc
                 if (!Storage._shutdownTokenSource.IsCancellationRequested)
                 {
                     using var _ = Storage._connectionPool.Target.GetPooledConnection(out var connection);
-                    if (TryGetDatabaseId(connection, key, out var dataId))
+                    try
```

Comment:
this is the critical part.  tryign to get the db id (which might write) was pulled out of ReadColumn.  Now readcolumn can be fully concurrent without worry about locking exceptions.